File: ui/Notifications/Dtos/NotificationDto.ts
import { CrudTimeDto } from "../../Helpers/DataGrid/CrudTimeDto";

export interface NotificationDto {
    id: number;
    readStatus: boolean;
    messageDto: MessageDto;
    userId: number;
    roleId: number;
    firstName: string;
    lastName: string;
    userName: string;
    roleName: string;
    updateStatus: boolean;
}

export interface MessageDto {
    header: string;
    body: string;
    footer: string;
}

export interface NotificationCrudStatusAndDataDto {
    crudStatus: string;
    notifications: NotificationDto[];
}

export const initialMessageDto:MessageDto={
    header: "",
    body: "",
    footer: ""
}

export const initialNotificationDto:NotificationDto={
    id: 0,
    readStatus: false,
    messageDto: initialMessageDto,
    userId: 0,
    roleId: 0,
    firstName: "",
    lastName: "",
    userName: "",
    roleName: "",
    updateStatus: false,
}

export interface Absentees extends CrudTimeDto {
    id: number;
    userName: string;
    lastName: string;
    userId: number;
    roleId: number;
    roleName: string;
    firstName: string;
    gender: string;
    email: string;
    password:string
}

File: api/Bussiness/Services/NotificationService/Dtos/NotificationDto.cs
using Bussiness.Helper.Dtos;
using System.Text.Json;

namespace Bussiness.Services.NotificationService.Dtos
{
    public class NotificationDto
    {
        public int Id { get; set; }
        public bool ReadStatus { get; set; }
        public JsonElement MessageDto { get; set; }
        public int UserId { get; set; }
        public int RoleId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string UserName { get; set; }
        public string RoleName { get; set; }
        public bool UpdateStatus { get; set; }
    }


    public class NotificationDto1
    {
        public int Id { get; set; }
        public bool ReadStatus { get; set; }
        public MessageDto MessageDto { get; set; }
        public int UserId { get; set; }
        public int RoleId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string UserName { get; set; }
        public string RoleName { get; set; }
        public bool UpdateStatus { get; set; }
    }
    public class MessageDto
    {
        public string Header { get; set; }
        public string Body { get; set; }
        public string Footer { get; set; }
    }
    public class NotificationCrudStatusAndDataDto
    {
        public string CrudStatus { get; set; }
        public List<NotificationDto> Notifications { get; set; }
    }

    public class MessageDto1
    {
        public string Header { get; set; }
        public Body Body { get; set; }
        public string Footer { get; set; }
    }
    public class Body
    {
        public List<Absentiess> Absentiesses { get; set; }
    }

    public class Absentiess:CrudTimeDto
    {
        public int Id { get; set; }
        public string UserName { get; set; }
        public string LastName { get; set; }
        public int UserId { get; set; }
        public int RoleId { get; set; }
        public string RoleName { get; set; }
        public string FirstName { get; set; }
        public string Gender { get; set; }
        public string Email { get; set; }
    }


}
